### ISSUE TITLE
Microsoft.AspNetCore.Identity.EntityFrameworkCore Integration

### ISSUE THREAD

**Issue by tim-elvidge** [simpleinjector/SimpleInjector#434]
Could the Integration Guide for AspNet Core Mvc be extended to cover Microsoft.AspNetCore.Identity.EntityFrameworkCore?

Having spent a few hours looking through StackOverflow and various other blogs I couldn't find an up to date example of how this can be done with SimpleInjector.

I expect that anyone trying to use SimpleInjector with Microsoft.AspNetCore.Identity.EntityFrameworkCore will do what I am doing now - giving up and going back to the Microsoft's DI as I can't afford the time to try and make this work.

**Comment by dotnetjunkie** [comment]
Can you describe what it is you like to achieve? What is the problem you are having?

**Comment by tim-elvidge** [comment]
@dotnetjunkie Thanks for the quick reply.
Basically just trying to replace Microsoft's DI with SimpleInjector. I have used SimpleInjector on MVC 5 projects with no issue but then Identity Framework there was not injected. I read through Identity framework causing trouble for non-conformers #328 and Question - Verify Fails in Asp.Net Core Startup - Creating the instance for type AccountController failed. No HttpContext #403 along with other blogs and can't for the life of me work out how to use register UserManager<ApplicationUser>, RoleManager<IdentityRole>, SignInManager<ApplicationUser>, IOptions<IdentityCookieOptions> etc.

So basically just wanting some example of how to get started. I'm pretty sure that I won't be the only one looking at how to do this. I tried cross-wiring but that didn't work. Would prefer  to do it without.

**Comment by dotnetjunkie** [comment]
I think that #328 actually has some good guidance to go with. See: https://github.com/simpleinjector/SimpleInjector/issues/328#issuecomment-261770313 .

The advice basically is: keep the Core DI in place for 3rd party libraries that depend on the Core DI infrastructure. Those libs typically depend too much on the expected details of that DI system that non-conformers just can't comply with. 

Do use Simple Injector to build up application components, and use the Core DI configuration system for ASP.NET Core and 3rd party stuff. This way you don't need an integration package for a 3rd party lib such as Entity Framework.

**Comment by tim-elvidge** [comment]
Ok in that case I'll just skip using SimpleInjector until someone else has got an example of how this might work. I imagine that most users requiring Microsoft.AspNetCore.Identity.EntityFrameworkCore will carry on using Microsoft's DI as there is just no documentation or examples on how to do it any other way. I'll try and see if StructureMap, Autofac etc. has any advice on this and if I can get it working will post back a gist of how it's done.

**Comment by dotnetjunkie** [comment]
Whatever works best fkr you. Glad to be of assistance.

**Comment by tim-elvidge** [comment]
> The advice basically is: keep the Core DI in place for 3rd party libraries that depend on the Core DI infrastructure. 
Does this mean you consider EntityFrameworkCore a 3rd party library that depends on the Core DI??

**Comment by dotnetjunkie** [comment]
Absolutely. I consider EF a 3rd party library. But even if you consider it a core component of .NET Core, my advice stays the same: cross-wire things that you need to depend upon in your application components and vice versa. For components such as EF, the number of cross-wirings you need are typically really low. I would even say that typical use doesn't require _any_ cross-wirings. I've been using EF for years and never needed to do these kinds of things at all. Perhaps your case is different, but in that case I am very interested in what you're trying to achieve.

**Comment by dotnetjunkie** [comment]
We made integrating with Identity Framework (and other 3rd parties) easier with this issue: https://github.com/simpleinjector/SimpleInjector/issues/437.

**Comment by tim-elvidge** [comment]
@dotnetjunkie, the fix and the documentation worked a treat.
